service/matching/handler/engine.go

```diff
@@ -539,6 +539,10 @@ pollLoop:
 		if err != nil {
 			return nil, fmt.Errorf("couldn't load tasklist namanger: %w", err)
 		}
+		pollerWaitTimeMs := int64(0)
+		if deadline, ok := pollerCtx.Deadline(); ok {
+			pollerWaitTimeMs = int64(time.Until(deadline).Milliseconds())
+		}
```

Comment:
maybe move this logic just before the empty response return. (line 564 before change)